apis/python/src/tiledbsoma/common.cc

```diff
@@ -229,7 +241,49 @@ void set_metadata(
     if (value_buffer.ndim != 1)
         throw py::type_error("Only 1D Numpy arrays can be stored as metadata");
 
-    auto value_num = is_tdb_str(value_type) ? value.nbytes() : value.size();
+    auto value_num = is_tdb_str(value_type) ?
```

Comment:
is it ever possible for value_type to be `== UTF8`?  UTF8 can have a variable number of bytes per character, so this will probably act strangely....

In theory, it would also be possible for numpy to generate an array with UTF-16 and other types.  Suggest either checking/throwing if you get an unexpected type, or adding tests and explicit handling.